src/haddock/modules/analysis/contactmap/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""Compute contacts and generate heatmaps."""
+from copy import deepcopy
+from pathlib import Path
+
+from haddock.core.typing import Any, FilePath
+from haddock.libs.libparallel import Scheduler
+from haddock.modules import BaseHaddockModule
+from haddock.modules.analysis.contactmap.contmap import (
+    ContactsMap,
+    ClusteredContactMap,
+    get_clusters_sets,
+    topX_models,
+    )
+
+
+RECIPE_PATH = Path(__file__).resolve().parent
+DEFAULT_CONFIG = Path(RECIPE_PATH, "defaults.yaml")
+
+
+class HaddockModule(BaseHaddockModule):
+    """HADDOCK3 module to compute complexes contacts and generate heatmap."""
+
+    name = RECIPE_PATH.name
+
+    def __init__(
+            self,
+            order: int,
+            path: Path,
+            *ignore: Any,
+            init_params: FilePath = DEFAULT_CONFIG,
+            **everything: Any,
+            ) -> None:
+        """Initialize class."""
+        super().__init__(order, path, init_params)
+
+    @classmethod
+    def confirm_installation(cls) -> None:
+        """Confirm if contact executable is compiled."""
+        return
+
+    def _run(self) -> None:
+        """Execute module."""
+        # Get the models generated in previous step
+        if type(self.previous_io) == iter:
+            _e = "This module cannot come after one that produced an iterable."
+            self.finish_with_error(_e)
+        try:
+            models = self.previous_io.retrieve_models(individualize=True)
+        except AttributeError as e:
+            self.finish_with_error(e)
+
+        # Initiate holder of all jobs to be run by the `Scheduler`
+        contact_jobs: list[Any] = []
```

Comment:
Only use `Any` if the data structure is polymorphic, the type can be `Union` of different types of jobs